sound/soc/sof/topology.c

```diff
@@ -1445,6 +1445,53 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 	return 0;
 }
 
+/**
+ * sof_comp_alloc - alloc buffer for a new component and initialize it
+ * @swidget: SOF widget contained extend data
+ * @ipc_size: passed in with basic component size, will be updated to
+ *  the final IPC message size (including extended data) after returned
```

Comment:
before returning but a more precise explanation will be "IPC payload size that will be updated depending on the size of ext data"